main.go

```diff
@@ -44,39 +55,112 @@ func NewConfig(configPath string) *Config {
 
 	// Start YAML decoding from file
 	if err := d.Decode(&config); err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
 
-	return config
+	return config, nil
 }
 
-// Function will be run before main()
-func init() {
-	// Looking for config flag
-	flag.StringVar(&configPath, "config", "config.yml", "path to config file")
+// ValidateConfigPath just makes sure that the path provided is a file that can be read
+func ValidateConfigPath(path string) error {
+	s, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+	if s.IsDir() {
+		return fmt.Errorf("'%s' is a directory, not a normal file", path)
+	}
+	return nil
 }
 
-func main() {
-	// Create new config instance
-	config := NewConfig(configPath)
+// ParseFlags will create and parse the CLI flags and return the path to be used elsewhere
+func ParseFlags() (string, error) {
+	// String that contains the configured configuration path
+	var configPath string
+
+	// Set up a CLI flag called "-config" to allow users to supply the configurationg file
+	flag.StringVar(&configPath, "config", "./config.yml", "path to config file")
+
+	// Acutally parse the flags
+	flag.Parse()
+
+	// Validate the path first
+	if err := ValidateConfigPath(configPath); err != nil {
+		return "", err
+	}
+
+	// Return the configuation path
+	return configPath, nil
+}
 
-	// Create router and define routes
+// NewRouter generates the router used in the HTTP Server
+func NewRouter() *http.ServeMux {
```

Comment:
again, rather than calling this inside of `main()`, make it it's own function to be tested. Right now, there is nothing that can really fail, but when things get crazier, it's worth doing.